quickwit/quickwit-indexing/src/source/ingest/mod.rs

```diff
@@ -432,13 +462,22 @@ impl Source for IngestSource {
 
         loop {
             match time::timeout_at(deadline, self.fetch_stream.next()).await {
-                Ok(Ok(fetch_payload)) => {
-                    self.process_fetch_response(&mut batch_builder, fetch_payload)?;
+                Ok(Ok(fetch_message)) => match fetch_message.message {
+                    Some(fetch_message::Message::Payload(fetch_payload)) => {
+                        self.process_fetch_payload(&mut batch_builder, fetch_payload)?;
 
-                    if batch_builder.num_bytes >= BATCH_NUM_BYTES_LIMIT {
-                        break;
+                        if batch_builder.num_bytes >= BATCH_NUM_BYTES_LIMIT {
+                            break;
+                        }
                     }
-                }
+                    Some(fetch_message::Message::Eof(fetch_eof)) => {
+                        self.process_fetch_eof(&mut batch_builder, fetch_eof)?;
+                    }
+                    None => {
```

Comment:
When does this happen? Is it normal behavior or not?

Reply:
It should never happen. It's a by-product of the `oneof`.